src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -605,32 +639,49 @@ private async Task ExecuteClusterTransactionOneByOne(List<ClusterTransactionComm
             }
         }
 
-        private void OnClusterTransactionCompletion(ClusterTransactionCommand.SingleClusterDatabaseCommand command,
-            BatchHandler.ClusterTransactionMergedCommand mergedCommands, Exception exception = null)
+        private bool OnClusterTransactionCompletion(ClusterTransactionCommand.SingleClusterDatabaseCommand command,
+            BatchHandler.ClusterTransactionMergedCommand mergedCommands, RemoveTask removeTask, Exception exception = null)
         {
+            var finished = true;
             try
             {
                 var index = command.Index;
+
+                LastCompletedClusterTransactionIndex = index;
```

Comment:
should we set if the cluster tx failed on the database level? (e.g. out of disk space)

Reply:
Moved